builder/rootfs/bin/boot

```diff
@@ -18,153 +18,9 @@ export ETCD_PORT=${ETCD_PORT:-4001}
-DOCKER_PID=$!
-
-# wait for docker to start
-while [[ ! -e /var/run/docker.sock ]]; do
-	sleep 1
-done
-
-# build required images if they don't already exist
-if ! docker history deis/slugbuilder >/dev/null 2>&1; then
-	docker build -t deis/slugbuilder /usr/local/src/slugbuilder/
-fi
-
-if ! docker history deis/slugrunner >/dev/null 2>&1; then
-	docker build -t deis/slugrunner /usr/local/src/slugrunner/
-fi
-
-function gen_host_keys {
-	if ! etcd_get sshHostKey; then
-		# generate the keys, then set them up in etcd
-		/usr/bin/ssh-keygen -A
-		for type in dsa ecdsa ed25519 rsa; do
-			cat "/etc/ssh/ssh_host_${type}_key" | etcd_set_default_stdin "sshHost${type}Key"
-			cat "/etc/ssh/ssh_host_${type}_key.pub" | etcd_set_default_stdin "sshHost${type}PubKey"
-		done
-		cat "etc/ssh/ssh_host_key" | etcd_set_default_stdin sshHostKey
-		cat "/etc/ssh/ssh_host_key.pub" | etcd_set_default_stdin sshHostPubKey
-	else
-		# pull the keys from etcd
-		for type in dsa ecdsa ed25519 rsa; do
-			etcd_get "sshHost${type}Key" > "/etc/ssh/ssh_host_${type}_key"
-			etcd_get "sshHost${type}PubKey" > "/etc/ssh/ssh_host_${type}_key.pub"
-		done
-		etcd_get sshHostKey > /etc/ssh/ssh_host_key
-		etcd_get sshHostPubKey > /etc/ssh/ssh_host_key.pub
-	fi
-	# set private key permissions to 0600
-	chmod 0600 /etc/ssh/ssh_host_*key
-}
-
-gen_host_keys
-
-# wait until confd generates the push-images script
-while [[ ! -e /usr/local/bin/push-images ]]; do
-	echo "waiting for confd to generate '/usr/local/bin/push-images'..."
-	sleep 1
-done
-
-# start an SSH daemon to process `git push` requests
-/usr/sbin/sshd -D -e &
-SSHD_PID=$!
-
-# start a cleanup script to remote old repositories and images
-/bin/cleanup &
-CLEANUP_PID=$!
-
-# smart shutdown on SIGINT and SIGTERM
-function on_exit() {
-	kill -TERM $DOCKER_PID $SSHD_PID $CLEANUP_PID
-	wait $DOCKER_PID $SSHD_PID $CLEANUP_PID 2>/dev/null
-	exit 0
-}
-trap on_exit INT TERM EXIT
-
-echo deis-builder running...
-
-# publish the service to etcd using the injected EXTERNAL_PORT
-if [[ ! -z $EXTERNAL_PORT ]]; then
-
-	# configure service discovery
-	PORT=${PORT:-22}
-	PROTO=${PROTO:-tcp}
-
-	set +e
-
-	# wait for the service to become available on PORT
-	sleep 1 && while [[ -z $(netstat -lnt | awk "\$6 == \"LISTEN\" && \$4 ~ \".$PORT\" && \$1 ~ \"$PROTO.?\"") ]] ; do sleep 1; done
-
-	# while the port is listening, publish to etcd
-	while [[ ! -z $(netstat -lnt | awk "\$6 == \"LISTEN\" && \$4 ~ \".$PORT\" && \$1 ~ \"$PROTO.?\"") ]] ; do
-		etcdctl --no-sync -C $ETCD set $ETCD_PATH/host $HOST --ttl $ETCD_TTL >/dev/null
-		etcdctl --no-sync -C $ETCD set $ETCD_PATH/port $EXTERNAL_PORT --ttl $ETCD_TTL >/dev/null
-		sleep $(($ETCD_TTL/2)) # sleep for half the TTL
-	done
-
-	# if the loop quits, something went wrong
-	exit 1
-
-fi
+echo External port $EXTERNAL_PORT
 
+builder
```

Comment:
As I look at that, I think we could do it pretty easily. I need to see whether there's anything magical in environment_proxy. But if not, it would be pretty straightforward. I'll test out the `exec` method right now, and also look at replacing the entire script.